power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -190,6 +190,7 @@ template <symmetry_tag sym_type> struct MathModelParam {
     std::vector<BranchCalcParam<sym>> branch_param;
     ComplexTensorVector<sym> shunt_param;
     ComplexTensorVector<sym> source_param;
+    std::vector<std::pair<DoubleComplex, DoubleComplex>> source_param_y0_y1;
```

Comment:
I am against the idea that we have two `source` stuff in `MathModelParam`. I think we need to change the way of math parameters of source in general. Concretely, I propose:

1. Define a new struct `SourceCalcParam` which is not a template.

```c++
struct SourceCalcParam {
    DoubleComplex y1;
    DoubleComplex y0;
};
```

2. Change the old `source_param` definition in `MathModelParam` to the new.

from 
```c++
    ComplexTensorVector<sym> source_param;
```

to
```c++
    std::vector<SourceCalcParam> source_param;
```

3. Rewrite the `template <symmetry_tag sym> ComplexTensor<sym> math_param()` in source to the new way of just exporting `y1` and `y0` instead of a scalar or tensor.

4. Rewrite the power flow solver functions where the source scalar/tensor is expected. Either directly use `y1` for symmetric calculation, or build source y tensor on the fly for asymmetric calculation. You can create an inline function in the `common_solver_functions.hpp` to build the y tensor matrix.

5. In the calculate source results, you can then directly use the `y1` and `y0`.